demos/classification_benchmark_demo/cpp_gapi/main.cpp

```diff
@@ -133,19 +162,12 @@ int main(int argc, char* argv[]) {
         });
 
         /** Configure network **/
+        auto nets = cv::gapi::networks();
         auto config = ConfigFactory::getUserConfig(FLAGS_d, FLAGS_nireq, FLAGS_nstreams, FLAGS_nthreads);
-        // clang-format off
-        const auto net =
-            cv::gapi::ie::Params<nets::Classification>{
-                FLAGS_m,  // path to topology IR
-                fileNameNoExt(FLAGS_m) + ".bin",  // path to weights
-                FLAGS_d  // device specifier
-            }.cfgNumRequests(config.maxAsyncRequests)
-             .pluginConfig(config.getLegacyConfig());
-        // clang-format on
-
+        execution_providers_t providers = createProvidersFromString(FLAGS_ep);
+        nets += util::create_execution_network<nets::Classification>(FLAGS_m, config, providers);
```

Comment:
Why did you introduce `createProvidersFromString()` and `create_execution_network()` separately? Merging them would reduce the number of ifdefs

Reply:
Modularity has drawbacks. For example:
> [Complexity, coordination, testing, and integration are some of the common issues that arise.](https://www.linkedin.com/advice/0/what-benefits-challenges-modular-system-design)

The point i brought in the original comment refers to complexity. Can you explain me why you preferred modularity over complexity here?

Reply:
I wouldn't like idea to discuss the pointless questions in review - like mixing up the concept of "System Architecture" (which reflected in the article) and the "Structured programming paradigm". 
By the way, according to the second paradigm - it'd better to have TWO functions: one for parsing arguments (like we have in multiple other demos here) and another one for using those arguments for creating functional entities.

Additionally, since we decided to introduce additional field 'backend' into arguments row it appeared that we are not required in MACRO anymore and such kind of "complexity" will gone

If you still do care about TWO functions instead of one I'm not concerned: let's unify it, because writing the demo, in my opinion, is not a place to carry out "profound architectural discussions"